src/language/ast.ts

```diff
@@ -63,15 +82,97 @@ export interface ASTKindToNode {
   ModelDescription: ModelDescriptionNode;
   Name: NameNode;
   OptionalName: OptionalNameNode;
+  EndpointNamespaceTypeDefinition: EndpointNamespaceTypeDefinitionNode;
+  EndpointVerb: EndpointVerbNode;
+  EndpointUrl: EndpointUrlNode;
+  EndpointTypeDefinition: EndpointTypeDefinitionNode;
+  EndpointResponse: EndpointResponseNode;
+  SecurityDefinition: SecurityDefinitionNode;
 }
 
 export interface DocumentNode {
   readonly kind: 'Document';
   readonly loc?: Location;
-  readonly definitions: ReadonlyArray<TypeDefinitionNode>;
+  readonly definitions: ReadonlyArray<TypeDefinitionNode | EndpointNamespaceTypeDefinitionNode>;
```

Comment:
Let's do next thing

```
TypeDefinitionNode = EndpointDomainTypeDefinitionNode | ModelDomainTypeDefinitionNode
```

Reply:
We can't do that because [here, at validation rules](https://github.com/frolovdev/anyspec/blob/master/src/validation/rules/knownTypeNames.rule.ts#L24) we expect `TypeDefinitionNode` to have `name` property and we don't have `name` at `EndpointNamespaceTypeDefinitionNode`. 
I'm not sure what to do in this case, I guess we just can hardcode a virtual name to each `EndpointNamespaceTypeDefinitionNode` when we parse it. Or change validation logic.